internal/remoteconfig/remoteconfig.go

```diff
@@ -243,12 +248,18 @@ func Subscribe(product string, callback ProductCallback, capabilities ...Capabil
 	if client == nil {
 		return ErrClientNotStarted
 	}
-	client.Lock()
-	defer client.Unlock()
+	client._productsMu.RLock()
+	defer client._productsMu.RUnlock()
 	if _, found := client.products[product]; found {
 		return fmt.Errorf("product %s already registered via RegisterProduct", product)
 	}
+
+	client._productsWithCallbacksMu.Lock()
+	defer client._productsWithCallbacksMu.Unlock()
 	client.productsWithCallbacks[product] = callback
+
+	client._capabilitiesMu.Lock()
+	defer client._capabilitiesMu.Unlock()
 	for _, cap := range capabilities {
 		client.capabilities[cap] = struct{}{}
```

Comment:
Those locks appear held longer than strictly necessary. Maybe that's not a big deal... but it feels like we might be able to improve that situation by using a sync'd `Map` type with the necessary accessor methods?

Reply:
This will be part of https://datadoghq.atlassian.net/browse/APPSEC-33536 👍 I also added a ref to your comment